cid.go

```diff
@@ -793,7 +793,7 @@ func CidFromReader(r io.Reader) (int, Cid, error) {
 	if cidLength > cap(br.dst) {
 		// If the multihash digest doesn't fit in our initial 64 bytes,
 		// efficiently extend the slice via append+make.
-		br.dst = append(br.dst, make([]byte, cidLength-cap(br.dst))...)
+		br.dst = append(br.dst, make([]byte, cidLength-len(br.dst))...)
```

Comment:
would it make sense to reuse `prefixLength` here?

Reply:
Hmmm. I think I prefer leaving it as `len` because it's clearer that we're appending to get `len` to get to `cidLength`. In these three lines of code, the only variable should be the final length/capacity we want, i.e. `cidLength`.